### PR TITLE
Add fastq_align_bwa subworkflow

### LINKED ISSUE #2412: new subworkflow: fastq_align_bwa
### Is there an existing subworkflow for this?

- [X] I have searched for the existing subworkflow

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this subworkflow, I added myself to the `Assignees` to facilitate tracking who is working on the subworkflow

### PR DESCRIPTION
**Pull request by JoseEspinosa**
Adding in a subworkflow for FASTQ ALIGN using BWA

## PR checklist

Closes #2412

- [x] This comment contains a description of changes (with reason).
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: subworkflows/nf-core/fastq_align_bwa/main.nf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// Alignment with BWA
+//
+
+include { BWA_MEM                 } from '../../../modules/nf-core/bwa/mem/main'
+include { BAM_SORT_STATS_SAMTOOLS } from '../bam_sort_stats_samtools/main'
+
+workflow FASTQ_ALIGN_BWA {
+    take:
+    ch_reads // channel (mandatory): [ val(meta), [ reads ] ]
+    ch_index // channel (mandatory): [ val(meta2, [ index ] ]
```
Comment (drpatelh) [inline subworkflows/nf-core/fastq_align_bwa/main.nf:11]: Add `ch_fasta` here too to pass to `BAM_SORT_STATS_SAMTOOLS` if required.

File: subworkflows/nf-core/fastq_align_bwa/main.nf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    //
+    // Map reads with BWA
+    //
+    BWA_MEM ( ch_reads, ch_index, sort_bam )
+    ch_versions = ch_versions.mix(BWA_MEM.out.versions.first())
+
+    //
+    // Sort, index BAM file and run samtools stats, flagstat and idxstats
+    //
+    BAM_SORT_STATS_SAMTOOLS ( BWA_MEM.out.bam, [] )
```
Comment (drpatelh) [inline subworkflows/nf-core/fastq_align_bwa/main.nf:27]:
```suggestion
    BAM_SORT_STATS_SAMTOOLS ( BWA_MEM.out.bam, ch_fasta )
```

File: subworkflows/nf-core/fastq_align_bwa/meta.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  - samtools/idxstats
+  - samtools/flagstat
+# TODO Update when we decide on a standard for subworkflow docs
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - reads:
```
Comment (drpatelh) [inline subworkflows/nf-core/fastq_align_bwa/meta.yml:22]:
```suggestion
  - ch_reads:
```

File: subworkflows/nf-core/fastq_align_bwa/meta.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      type: file
+      description: |
+        List of input FastQ files of size 1 and 2 for single-end and paired-end data,
+        respectively.
+  - index:
```
Comment (drpatelh) [inline subworkflows/nf-core/fastq_align_bwa/meta.yml:27]:
```suggestion
  - ch_index:
```

File: subworkflows/nf-core/fastq_align_bwa/main.nf (hunk 1 above)
Comment (JoseEspinosa) [reply subworkflows/nf-core/fastq_align_bwa/main.nf:11]: Addressed [ff530ff](https://github.com/nf-core/modules/pull/2451/commits/ff530ff3cf56818d22f32d93b1123bbbc4462f28)

File: subworkflows/nf-core/fastq_align_bwa/meta.yml (hunk 3 above)
Comment (JoseEspinosa) [reply subworkflows/nf-core/fastq_align_bwa/meta.yml]: Addressed [ff530ff](https://github.com/nf-core/modules/pull/2451/commits/ff530ff3cf56818d22f32d93b1123bbbc4462f28)

File: subworkflows/nf-core/fastq_align_bwa/meta.yml (hunk 4 above)
Comment (JoseEspinosa) [reply subworkflows/nf-core/fastq_align_bwa/meta.yml]: Addressed [ff530ff](https://github.com/nf-core/modules/pull/2451/commits/ff530ff3cf56818d22f32d93b1123bbbc4462f28)

**Review by Emiller88 — APPROVED** [review]
Looks great to me! Can't wait to `nf-core subworkflows install` it!

File: subworkflows/nf-core/fastq_align_star/meta.yml [hunk 5]
```diff
@@ -18,15 +18,34 @@ input:
...
+      pattern: "true or false"
+  - seq_platform:
+      type: string
+      description: |
+        Sequencing platform to be added to the bam header using the --outSAMattrRGline flag
+  - seq_center:
+      type: string
+      description: |
+        Sequencing center to be added to the bam header using the --outSAMattrRGline flag
+
```
Comment (drpatelh) [inline subworkflows/nf-core/fastq_align_star/meta.yml:48]: `ch_fasta` is missing here?

File: tests/subworkflows/nf-core/fastq_align_star/main.nf [hunk 6]
```diff
@@ -18,7 +18,7 @@ workflow test_align_star_single_end {
     seq_center = false
 
     STAR_GENOMEGENERATE ( fasta, gtf )
-    FASTQ_ALIGN_STAR ( input, STAR_GENOMEGENERATE.out.index, [], star_ignore_sjdbgtf, seq_platform, seq_center)
+    FASTQ_ALIGN_STAR ( input, STAR_GENOMEGENERATE.out.index, [ ], star_ignore_sjdbgtf, seq_platform, seq_center, [ ] )
```
Comment (drpatelh) [inline tests/subworkflows/nf-core/fastq_align_star/main.nf:21]:
```suggestion
    FASTQ_ALIGN_STAR ( input, STAR_GENOMEGENERATE.out.index, [], star_ignore_sjdbgtf, seq_platform, seq_center, [] )
```

File: tests/subworkflows/nf-core/fastq_align_star/main.nf [hunk 7]
```diff
@@ -36,6 +36,6 @@ workflow test_align_star_paired_end {
...
```
Comment (drpatelh) [inline tests/subworkflows/nf-core/fastq_align_star/main.nf:39]:
```suggestion
    FASTQ_ALIGN_STAR ( input, STAR_GENOMEGENERATE.out.index, [], star_ignore_sjdbgtf, seq_platform, seq_center, [] )
```

**Review by drpatelh — APPROVED** [review]
LGTM! Some minor changes but feel free to merge when these have been fixed and tests have passed.

File: subworkflows/nf-core/fastq_align_chromap/main.nf [hunk 8]
```diff
@@ -27,7 +27,7 @@ workflow FASTQ_ALIGN_CHROMAP {
     //
     // Sort, index BAM file and run samtools stats, flagstat and idxstats
     //
-    BAM_SORT_STATS_SAMTOOLS(CHROMAP_CHROMAP.out.bam, [])
+    BAM_SORT_STATS_SAMTOOLS(CHROMAP_CHROMAP.out.bam, ch_fasta[1])
```
[the remark this answers is not in the document]
Comment (JoseEspinosa) [reply subworkflows/nf-core/fastq_align_chromap/main.nf]: Addressed here [210e185](https://github.com/nf-core/modules/pull/2451/commits/210e185d9ba56450b0dbbf36d0fb0631b9018801)

**Pull request merged**